shared/src/simulation/activities/reassign-treatments.ts

```diff
@@ -610,38 +671,269 @@ function assignTreatments(
+                requiredByType['red']!,
+                requiredByType['yellow']!
+            )
+        ),
+        // Green should not be mixed. Therefore we ceil them before merging.
+        ceilResourceDescription(requiredByType['green']!)
+    );
+}
+
+function calculateMissingPersonnel(
+    groupedPatients: { [Key in TreatablePatientStatus]?: Patient[] },
+    personnel: CateringPersonnel[],
+    personnelTreatments: PersonnelToPatientCategoryDict,
+    patientTreatments: PatientToPersonnelDict
+) {
+    const cateringPersonnelDict = Object.fromEntries(
+        personnel.map((cateringPersonnel) => [
+            cateringPersonnel.personnel.id,
+            cateringPersonnel,
+        ])
+    );
+
+    const patientStatusMissingPersonnel = Object.fromEntries(
+        (['red', 'yellow', 'green'] as const).map((patientStatus) => {
+            const patients = groupedPatients[patientStatus] ?? [];
+
+            let statusMissingPersonnel =
+                PersonnelResource.create().personnelCounts;
+            patients.forEach((patient) => {
+                const patientMissingPersonnel = cloneDeepMutable(
+                    minimumRequiredPersonnel[patientStatus]
+                );
+
+                personnelTypePriorityList.forEach((personnelType) => {
+                    (patientTreatments[patient.id]?.[personnelType] ?? [])
+                        .map(
+                            (personnelId) => cateringPersonnelDict[personnelId]
+                        )
+                        .forEach((p) => {
+                            if (p) {
+                                patientMissingPersonnel[personnelType] -=
+                                    1 /
+                                    (requiresExclusivity[patientStatus]?.[
+                                        personnelType
+                                    ]
+                                        ? sumOfTreatments(p)
+                                        : p.catersFor[patientStatus]);
+                            }
+                        });
+                });
+                statusMissingPersonnel = addResourceDescription(
+                    statusMissingPersonnel,
+                    maxResourceDescription(patientMissingPersonnel, 0)
+                );
+            });
+            return [patientStatus, statusMissingPersonnel] as const;
+        })
+    ) as { [K in TreatablePatientStatus]: ResourceDescription<PersonnelType> };
+
+    const totalMissingPersonnel = addResourceDescription(
+        ceilResourceDescription(
+            addResourceDescription(
+                patientStatusMissingPersonnel.red,
+                patientStatusMissingPersonnel.yellow
+            )
+        ),
+        ceilResourceDescription(patientStatusMissingPersonnel.green)
+    );
+
+    totalMissingPersonnel.notarzt = Math.max(
+        patientStatusMissingPersonnel.red.notarzt,
+        patientStatusMissingPersonnel.yellow.notarzt
+    );
```

Comment:
Why do we only need the maximum and not the sum?

Reply:
This is because the `patientStatusMissingPersonnel` describes how much personnel we need to secure all patients with this status. So for 2 red and 2 yellow patients, `patientStatusMissingPersonnel.red.notarzt` and `patientStatusMissingPersonnel.yellow.notarzt` each have value `1` because one `notarzt` cannot treat more than 2 red or more than 2 yellow patients. If we had more of any of these types, we would require more `notarzt`.

However, because one `notarzt` can treat 2 red *and* 2 yellow patients , in the case above we do not need `1 + 1 = 2` notarzt, `Math.max(1, 1) = 1` is sufficient. In general, if some personnel has the logical operator `and`, we need to use `Math.max` to combine required personnel of this type across categories and if  the logical operator is `or`, we need to use `+`.

Reply:
This creates a pretty tight coupling between this code and the personnel templates which may be changed by an imported exercise.

But I don't know a better solution at the moment...

Reply:
The entire code is strongly coupled to the templates anyways because they can define own personnel types which are then missing in `personnelPrioities` and friends, they can omit the types required in the treatment logic, are partially ignored by the 'no more than two patients per ordinary personnel' rules etc. If we wanted to build clean code with properly configurable templates, we would need to scrap the majority of treatment code and spend much energy on improving the overall capacity / need model.